src-java/flowmonitoring-topology/flowmonitoring-storm-topology/src/main/java/org/openkilda/wfm/topology/flowmonitoring/service/FlowCacheService.java

```diff
@@ -93,6 +113,16 @@ public void updateFlowInfo(UpdateFlowCommand info) {
         flowStates.put(info.getFlowId(), FlowMapper.INSTANCE.toFlowState(info));
     }
 
+    /**
+     * Update HA-Flow info.
+     */
+    public void updateHaFlowInfo(UpdateHaSubFlowCommand info) {
+        HaSubFlow haSubFlow = haSubFlowRepository.findById(info.getFlowId())
+                .orElseThrow(() -> new IllegalStateException(format("HA-Sub-Flow %s is not found.",
+                        info.getFlowId())));
```

Comment:
Who will handle this exception? We will get something like "unhanded exception". Look at `isIncompleteHaFlow`. Do the same and just log an error.